app/views/articles/_sidebar.html.erb

```diff
@@ -19,16 +19,16 @@
         </div>
       <% end %>
     <% end %>
-    <% cache("display-area-left-#{rand(5)}", expires_in: 5.minutes) do %>
-      <% @left_sidebar_ad = DisplayAd.for_display("sidebar_left") %>
+    <% cache("display-area-left-#{rand(5)}-#{user_signed_in?}", expires_in: 5.minutes) do %>
```

Comment:
I've added this cache key of user_signed_in because we need to reload the display ads based on whether the ad should appear to a logged in or logged out user.

Reply:
Really great job being thorough with all of this!